### ISSUE TITLE
[Bug] Fix PDAL support for OSGeo4W

### ISSUE THREAD

**Issue by ninsbl** [OSGeo/grass#1821]
**Describe the bug**
OSGeo4W ships PDAL version 2.3.
When PDAL support is activated for OSGeo4W with a path to pdal-config, pdal-config is found, but GRASS complains that the PDAL library is not suitable `Unable to locate suitable (>=1.7.1) PDAL library`.

**To Reproduce**
See: https://github.com/ninsbl/grass/runs/3408602894?check_suite_focus=true

**Expected behavior**
It is important to have PDAL support on MS Windows too, so GRASS should be compiled with PDAL support on MS Windows as well.
Would be great to have that working for 8.0 but I set the 8.2 milestone. Feel free to change if feasible to get it into 8.0...

**System description (please complete the following information):**
GRASS OSGeo4W CI
GRASS GIS version 7.9 / 8.0 preview

**Additional context**
Here is how the PDAL packages for OSGeo4W are built:
https://github.com/jef-n/OSGeo4W/blob/master/src/pdal/osgeo4w/package.sh

**Comment by marisn** [comment]
It just means compilation and linking to PDAL library has failed:
https://github.com/OSGeo/grass/blob/9f853f0d3a396879d01f4a39a9769efc09e5b6d7/configure.in#L1020

Actual error is visible in the configure.log file. Content of that file might help to shed some light into the issue. As usual – if Windows users need PDAL support in GRASS, one of Windows users must step up and fix the issue. I could look into the issue if a link to a VM image with suitable environment is provided.

**Comment by ninsbl** [comment]
Fair enough. Here: https://github.com/ninsbl/grass/runs/3412192862?check_suite_focus=true is at least the config.log

**Comment by ninsbl** [comment]
This likely depends on: #348
See also:
https://github.com/jef-n/OSGeo4W/pull/3

**Comment by HuidaeCho** [comment]
Yes, unfortunately, OSGeo4W PDAL seems to be compiled with MSVC and there is no way to link against it using GCC.
```
configure:9427: x86_64-w64-mingw32-g++ -o conftest.exe -g -O2   -I/c/OSGeo4W/include -O2 -fPIC -std=c++11 -I/c/OSGeo4W/include -I/c/OSGeo4W/include -I/c/OSGeo4W/include/libxml2   -Wl,--export-dynamic,--enable-runtime-pseudo-reloc  -L/c/OSGeo4W/lib conftest.cpp  -L/c/OSGeo4W/bin -lpdalcpp >&5
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: warning: --export-dynamic is not supported for PE+ targets, did you mean --export-all-symbols?
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\msys64\tmp\ccxle6JS.o: in function `pdal::SimplePointTable::SimplePointTable(pdal::PointLayout&)':
```
Alternatively, using mingw-w64-x86_64-pdal seems to be overkill because it requires *additional* 108 packages, but if you *really* need it, you may try it.
![image](https://user-images.githubusercontent.com/7456117/208162542-9f51aec7-c3a4-4bb9-b078-1ec85833405f.png)
